### ISSUE TITLE
[BUG] SetTags/SetTagsAsync methods could conflict with Azure Policy 

### ISSUE THREAD

**Issue by J0F3** [Azure/azure-sdk-for-net#33997]
### Library name and version

Azure.ResourceManager 1.3.2, Azure.ResourceManager.Storage 1.1.0, and a lot more (if not all)

### Describe the bug

If there is an Azure Policy in place which enforces the presence of certain Tag names the `SetTags` resp. `SetTagsAsync` methods will not work. Potentially this is the case for all resources which have a `SetTags` resp. `SetTagsAsync` method. I have tested it with GenericResource, ResourceGroups and StorageAccount. An they have all the identically issue.

When I was looking into what the methods are actually doing I saw quickly that first the wohle set of tag gets deleted prior to set the new Tags. An that is exactly what lead to the problem. Basically it is this line of code:
```c#
 GetTagResource().Delete(WaitUntil.Completed, cancellationToken: cancellationToken);
```

The problem with that is, with the Azure Policy in place, you would get a Policy violation error here. Because deleting the wohle set of Tags would no be allowed by the Policy because it would also delete the Tags which would be required.

For me it is actually also not clear why this is done this way at all because the `TagResouce` has also a `CreateOrUpdate` method which can create or replace the whole set of Tag in one call. (see also: https://learn.microsoft.com/en-us/rest/api/resources/tags/create-or-update-at-scope?tabs=HTTP)

So to use directly the `CreateOrUpdate` method from the `TagResource` class instead the `SetTags` method of the resource specific class is also a workaround for the problem.

### Expected behavior

It should be possible to set tag to a resource with the `SetTags` resp. `SetTagsAsync` methods even when a Azure Policy which enforces certain Tags is in place. (As long the required Tags would actually get set.).

### Actual behavior

The method throws an error and tag get not set on to the resource / Resource Group.

### Reproduction Steps

1. Assign the Built-in Azure Policy "Require a tag on resources" or "Require a tag on resource groups" to a Resource Group to enforce the existence of a tag on Resources or the Resource Group.
2. Try to set tags (including the tag defined in the Policy assignment) to a Resource or the Resource Group itself with the corresponding `SetTags` resp. `SetTagsAsync` method.
3. `SetTags` / `SetTagsAsync` fails with an Policy violation error an the Tag are not set on the Resource / Resource Group.

Example code for Resource Group:
```c#
using Azure.Core;
using Azure.Identity;
using Azure.ResourceManager;
using Azure.ResourceManager.Resources;

internal class AzureTags
{
    private static void Main()
    {

        const string resourceGroupId = "/subscriptions/d67c6ec7-b609-4789-be30-02ee131266b6/resourceGroups/testrg";

        IDictionary<string, string> tagsToSet =
           new Dictionary<string, string>()
           {
               {"CostCenter", "required by policy" },
               {"otherTag", "otherValue" },
           };


        ArmClient armClient = new ArmClient(new DefaultAzureCredential());

        GenericResource resourceGroup = armClient.GetGenericResource(new ResourceIdentifier(resourceGroupId));

        resourceGroup.SetTags(tagsToSet);
    }
}
```

Request fails even tough it should succeeds because the Tag required by the policy is get set:
> An unhandled exception of type 'Azure.RequestFailedException' occurred in Azure.ResourceManager.dll: 'Resource 'testrg' was disallowed by policy. Policy identifiers: ...
> Status: 403 (Forbidden)
> ErrorCode: RequestDisallowedByPolicy
> 

Example code for Storage Account:
```c#
using Azure.Core;
using Azure.Identity;
using Azure.ResourceManager;
using Azure.ResourceManager.Storage;
using Azure.ResourceManager.Resources;

internal class AzureTags
{
    private static void Main()
    {

        const string subscriptionId = "/subscriptions/d67c6ec7-b609-4789-be30-02ee131266b6";
        const string rgName = "testrg";
        const string storageAccountName = "testaccount";

        IDictionary<string, string> tagsToSet =
           new Dictionary<string, string>()
           {
               {"CostCenter", "must be present" },
               {"otherTag", "otherValue" },
           };


        ArmClient armClient = new ArmClient(new DefaultAzureCredential());

        SubscriptionResource subscription = armClient.GetSubscriptionResource(new ResourceIdentifier(subscriptionId));

        ResourceGroupResource rg = subscription.GetResourceGroup(new ResourceIdentifier(rgName));

        StorageAccountResource storageAccount = rg.GetStorageAccount(storageAccountName);

        storageAccount.SetTags(tagsToSet);
    }
}
```

Basically same `RequestDisallowedByPolicy` exception as above.

### Environment

N/A

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by xboxeer** [comment]
@J0F3  this could be a bug, we'll internally triage this and decide when and how we will provide a fix to it

**Comment by gmantri** [comment]
BTW, same issue can also happen if there is a management lock on the resource. I posted a similar issue - https://github.com/Azure/azure-sdk-for-net/issues/35290.

**Comment by J0F3** [comment]
@jsquire @xboxeer @fengzhou-msft @live1206 
What is the status of this? The issue is open for over a year now without any progress. Do you have any intension to fix that one day?
A short update would be much appreciated. 😊
Thx!

**Comment by jsquire** [comment]
@J0F3:  Unfortunately, I'm just initial triage for this; I have no insight into the current status.  We'd need an update from @live1206 or @ArthurMa1978.
